### PR TITLE
Move cuda_device to the constructor

### PR DESCRIPTION
**Pull request by scott-vsi**

### CODE REVIEW

File: python/pyface3d/pyface3d.cxx [hunk 1]
```diff
@@ -963,7 +963,7 @@ PYBIND11_MODULE(face3d, m)
 #endif
 
   py::class_<face3d::vertex_localizer>(m, "vertex_localizer")
-    .def(py::init<face3d::triangle_mesh const&>())
+    .def(py::init<face3d::triangle_mesh const&, int>())
```
Comment (decrispell) [inline python/pyface3d/pyface3d.cxx]: do you need to add a default argument here to avoid breaking existing python code?

Comment (scott-vsi) [reply python/pyface3d/pyface3d.cxx]: added

File: python/pyface3d/pyface3d.cxx [hunk 2]
```diff
@@ -963,7 +963,7 @@ PYBIND11_MODULE(face3d, m)
 #endif
 
   py::class_<face3d::vertex_localizer>(m, "vertex_localizer")
-    .def(py::init<face3d::triangle_mesh const&>())
+    .def(py::init<face3d::triangle_mesh const&, int>(), py::arg("semantic_map"), py::arg("cuda_device")=0)
```
Comment (decrispell) [inline python/pyface3d/pyface3d.cxx]: the first argument should be the mesh whose vertices you're going to be localizing, not the semantic map.  (The semantic_map gets passed into the __call__ function)

Comment (scott-vsi) [reply python/pyface3d/pyface3d.cxx]: fixed

**Review by decrispell — APPROVED** [review]

**Pull request merged**